ADAL/tests/unit/ADTokenCacheItemTests.m

```diff
@@ -43,110 +43,480 @@ - (void)tearDown
+}
+
+- (void)testIsExpired_whenExpiresOnIsOneHourFromNow_shoudReturnFalse
+{
+    ADTokenCacheItem *item = [self adCreateCacheItem:@"eric@contoso.com"];
     item.expiresOn = [NSDate dateWithTimeIntervalSinceNow:3600];
-    XCTAssertTrue(!item.isExpired, "The item is good for one more hour!");
+    
+    XCTAssertFalse(item.isExpired);
 }
-- (void)testIsEmptyUser
+
+#pragma mark - isEmptyUser
+
+- (void)testIsEmptyUser_whenUserInformationIsValid_shouldReturnFalse
 {
-    ADTokenCacheItem* item = [self adCreateCacheItem:@"eric@contoso.com"];
+    ADTokenCacheItem *item = [self adCreateCacheItem:@"eric@contoso.com"];
+    
     XCTAssertFalse(item.isEmptyUser);
+}
+
+- (void)testIsEmptyUser_whenUserInformationIsNil_shouldReturnTrue
+{
+    ADTokenCacheItem *item = [self adCreateCacheItem:@"eric@contoso.com"];
     item.userInformation = nil;
+    
     XCTAssertTrue(item.isEmptyUser);
-    item = [ADTokenCacheItem new];
-    XCTAssertTrue(item.isEmptyUser, "The default item should not have a user.");
 }
 
-- (void)verifySameUser:(NSString *)userId1
-               userId2:(NSString *)userId2
+#pragma mark - userInformation
+
+- (void)testUserInformation_byDefault_shouldBeNil
 {
-    ADTokenCacheItem* item1 = [self adCreateCacheItem:userId1];
-    ADTokenCacheItem* item2 = [self adCreateCacheItem:userId2];
+    ADTokenCacheItem *item = [ADTokenCacheItem new];
     
-    XCTAssertTrue([item1 isSameUser:item2], "Should be the same: '%@' and '%@", userId1, userId2);
-    XCTAssertTrue([item2 isSameUser:item1], "Should be the same: '%@' and '%@", userId1, userId2);
+    XCTAssertNil(item.userInformation);
 }
 
--(void)testIsSameUser
+#pragma mark - isSameUser
+
+- (void)testIsSameUser_whenOtherUserIdIsIdenticalToCurrent_shouldReturnTrue
 {
-    [self verifySameUser:nil userId2:nil];
-    [self verifySameUser:@" test user" userId2:@"test user"];
-    [self verifySameUser:@" test user  " userId2:@"     test user     "];
-    [self verifySameUser:@" test user" userId2:@"test user     "];
-    [self verifySameUser:@"test user" userId2:@"test user"];
+    ADTokenCacheItem *item1 = [ADTokenCacheItem new];
+    item1.userInformation = [self adCreateUserInformation:@"test user"];
+    ADTokenCacheItem *item2 = [ADTokenCacheItem new];
+    item2.userInformation = [self adCreateUserInformation:@"test user"];
+    
+    XCTAssertTrue([item1 isSameUser:item2]);
+    XCTAssertTrue([item2 isSameUser:item1]);
 }
 
--(void) testMultiRefreshTokens
+- (void)testIsSameUser_whenOtherUserNilCurrentUserNil_shouldReturnTrue
+{
+    ADTokenCacheItem *item1 = [ADTokenCacheItem new];
+    item1.userInformation = nil;
+    ADTokenCacheItem *item2 = [ADTokenCacheItem new];
+    item2.userInformation = nil;
+    
+    XCTAssertTrue([item1 isSameUser:item2]);
+    XCTAssertTrue([item2 isSameUser:item1]);
+}
+
+// TODO: do we need this behaviour?
+- (void)DISABLEDtestIsSameUser_whenOtherUserIdStartsWithSpaceButOtherPartIsIdenticalToCurrent_shouldReturnTrue
+{
```

Comment:
it was disabled because "isEqual" logic is changed and that test was failing. UserId was passing inside "allClaims" dictionary without trimming, and as a result 2 UserInfo objects were are not equal. Since we decided to remove "IsSameUser" method, that test is removed as well. Testing of UserInfo equality is moved to ADUserInformationTests.